### PR TITLE
Adding Google's gemma Model

### PR DESCRIPTION
**Pull request by monk1337**
Adding Gemma model config
https://huggingface.co/google/gemma-7b

Testing and working!

Screenshot 2024-02-21 at 9 22 07 PM

### CODE REVIEW

File: examples/gemma/gemma.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+flash_attention: true
+ 
+warmup_ratio: 0.5
+evals_per_epoch: 4
+eval_table_size:
+eval_max_new_tokens: 128
+saves_per_epoch: 1
+debug:
+#default deepspeed, can use more aggresive if needed like zero2, zero3
+deepspeed: deepspeed_configs/zero1.json
```
Comment (NanoCode012) [inline examples/gemma/gemma.yml]:
```suggestion
deepspeed:
```

File: examples/gemma/gemma.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+tokenizer_type: AutoTokenizer
+ 
+load_in_8bit: false
+load_in_4bit: true
+strict: false
+ 
+datasets:
+  - path: data.json
+    ds_type: json
+    type: alpaca
```
Comment (NanoCode012) [inline examples/gemma/gemma.yml]: Please adjust similar to other example files.

File: examples/gemma/gemma.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+val_set_size: 0.1
+output_dir: ./out
+ 
+adapter: qlora
+lora_r: 32
+lora_alpha: 16
+lora_dropout: 0.05
+lora_target_linear: true
+ 
+sequence_len: 4000
```
Comment (NanoCode012) [inline examples/gemma/gemma.yml]:
```suggestion
sequence_len: 4096
```

File: examples/gemma/gemma.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# use google/gemma-7b if you have access
+base_model: mhenrichsen/gemma-7b 
+model_type: AutoModelForCausalLM
+tokenizer_type: AutoTokenizer
+ 
+load_in_8bit: false
+load_in_4bit: true
+strict: false
+ 
+datasets:
```
Comment (mhenrichsen) [inline examples/gemma/gemma.yml]: change to something from hf. A lot of the examples use mhenrichsen/alpaca_2k_test

File: examples/gemma/gemma.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+lora_r: 32
+lora_alpha: 16
+lora_dropout: 0.05
+lora_target_linear: true
+ 
+sequence_len: 4000
+sample_packing: false
+pad_to_sequence_len: false
+ 
+wandb_project: axolotl
```
Comment (NanoCode012) [inline examples/gemma/gemma.yml]:
```suggestion
wandb_project:
```

File: examples/gemma/gemma.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+ 
+gradient_checkpointing: true
+early_stopping_patience:
+resume_from_checkpoint:
+local_rank:
+logging_steps: 1
+xformers_attention:
+flash_attention: true
+ 
+warmup_ratio: 0.5
```
Comment (NanoCode012) [inline examples/gemma/gemma.yml]:
```suggestion
warmup_ratio: 0.1
```

File: examples/gemma/gemma.yml (hunk 1 above)
Comment (mhenrichsen) [inline examples/gemma/gemma.yml]: deepspeed is not needed for an example, and should probably be deleted.

**Review by NanoCode012 — COMMENTED** [review]
Also, please rename file to `gemma/qlora.yml`

File: examples/gemma/gemma.yml (hunk 3 above)
Comment (mhenrichsen) [inline examples/gemma/gemma.yml]: the default sequence length of gemma is 8192.

**Review by mhenrichsen — APPROVED** [review]

**Pull request merged**